--- a/chrome/browser/ui/views/location_bar/BUILD.gn
+++ b/chrome/browser/ui/views/location_bar/BUILD.gn
@@ -50,6 +50,8 @@ source_set("location_bar") {
     "permission_chip_view.h",
     "permission_dashboard_view.cc",
     "permission_dashboard_view.h",
+    "profile_badge_view.cc",
+    "profile_badge_view.h",
     "selected_keyword_view.cc",
     "selected_keyword_view.h",
     "star_view.cc",
