diff --git a/dom/base/Navigator.cpp b/dom/base/Navigator.cpp
index 0000000000..1111111111 100644
--- a/dom/base/Navigator.cpp
+++ b/dom/base/Navigator.cpp
@@ -7,6 +7,7 @@
 // Needs to be first.
 #include "Navigator.h"

+#include "MaskConfig.hpp"
 #include "Geolocation.h"
 #include "base/basictypes.h"
 #include "mozilla/Components.h"
@@ -449,6 +450,11 @@
 void Navigator::GetOscpu(nsAString& aOSCPU, CallerType aCallerType,
                          ErrorResult& aRv) const {
   if (aCallerType != CallerType::System) {
+    // MaskConfig override for oscpu
+    if (auto value = MaskConfig::GetString("navigator.oscpu")) {
+      aOSCPU = NS_ConvertUTF8toUTF16(value.value());
+      return;
+    }
     // If fingerprinting resistance is on, we will spoof this value. See
     // nsRFPService.h for details about spoofed values.
     if (nsContentUtils::ShouldResistFingerprinting(GetDocShell(),
@@ -689,6 +695,8 @@
 }

 uint64_t Navigator::HardwareConcurrency() {
+  if (auto value = MaskConfig::GetUint64("navigator.hardwareConcurrency"))
+    return value.value();
   workerinternals::RuntimeService* rts =
       workerinternals::RuntimeService::GetOrCreateService();
   if (!rts) {
